### PR TITLE
fix(android)(8_0_X): APK signing will fail when using MD5 keystore and JDK 8+

### PR DESCRIPTION
**Pull request by jquick-axway**
**JIRA:**  https://jira.appcelerator.org/browse/TIMOB-26895

**Cherry-pick of PR:** #10888

### CODE REVIEW

File: android/cli/commands/_build.js [hunk 1]
```diff
@@ -798,16 +798,19 @@ AndroidBuilder.prototype.config = function config(logger, config, cli) {
 											_t.keystoreAliases.pop();
 										}
 
+										// Parse the keystore's alias name and signature algorithm.
+										// Note: Algorithm can return "MD5withRSA (weak)" on JDK 8 and higher.
+										//       Only extract 1st token since we need a valid algorithm name.
 										const aliasRegExp = /Alias name: (.+)/,
-											sigalgRegExp = /Signature algorithm name: (.+)/;
+											sigalgRegExp = /Signature algorithm name: (.[^\s]+)/;
```
Comment (sgtcoolguy) [inline android/cli/commands/_build.js]: that regex looks a little funky in that group. It went from saying: 'one or more of any character' to: 'one of any character followed by one or more non-spaces'. It probably works fine but may be a little misleading.

**Review by sgtcoolguy — APPROVED** [review]

File: android/cli/commands/_build.js (hunk 1 above)
Comment (jquick-axway) [reply android/cli/commands/_build.js]: Do you have a better solution? Your regex kung-fu is likely better than mine.
(I prefer to parse the old-school way via `indexOf()`.)

I tested my regex with the following script locally...
```javascript
var regex = /Signature algorithm name: (.[^\s]+)/;
//var text = "Signature algorithm name: MD5withRSA";
//var text = "Signature algorithm name:  MD5withRSA";
var text = "Signature algorithm name: MD5withRSA (weak)";
//var text = "Signature algorithm name: MD5withRSA  ( weak )  ";
//var text = "Signature algorithm name:  MD5withRSA  ( weak )  ";
//var text = "Signature algorithm name: SHA256withRSA";
//var text = "Signature algorithm name: ";
text = "The text above.\n" + text + "\nThe text below.\n";
var result = text.match(regex);
console.log(JSON.stringify(result));
```

**Comment by keerthi1032** [comment]
FR passed. Apk built successfully with MD5,SHA256,SHA1 keystone.

Test Environment:
Operating System
Name = Mac OS X
Version = 10.13.6
Architecture = 64bit
Node.js
Node.js Version = 8.9.1
npm Version = 5.5.1
Titanium CLI
CLI Version = 5.1.1
Titanium SDK
SDK Version = local sdk 8.0.1.v20190515105744
CLI =7.0.11
Studio =5.1.2.201903111843

**Comment by keerthi1032** [comment]
Jenkin is failing .npt able to merge

**Pull request merged**